packages/algolia/src/indexes/research-output.ts

```diff
@@ -0,0 +1,49 @@
+import { WaitablePromise } from '@algolia/client-common';
+import {
+  DeleteResponse,
+  SaveObjectResponse,
+  SearchOptions,
+} from '@algolia/client-search';
+
+import { ResearchOutputResponse } from '@asap-hub/model';
+import { SearchIndex } from './search-index';
+import { SearchResponse, SearchResponseEntity } from '../types/response';
+
+export type ResearchOutputSearchResponse = SearchResponse<
+  ResearchOutputResponse,
+  'research-output'
+>;
+export type ResearchOutputSearchResponseEntity = SearchResponseEntity<
+  ResearchOutputResponse,
+  'research-output'
+>;
+
+export class ResearchOutputSearchIndex extends SearchIndex<
+  ResearchOutputResponse,
+  'research-output'
+> {
+  public remove(entityId: string): Readonly<WaitablePromise<DeleteResponse>> {
+    return this.deleteObject(entityId);
+  }
+
+  public save(
+    researchOutput: ResearchOutputResponse,
+  ): Readonly<WaitablePromise<SaveObjectResponse>> {
+    return this.saveObject(researchOutput.id, researchOutput);
+  }
+
+  public async search(
+    query: string,
+    requestOptions?: SearchOptions,
+  ): Promise<ResearchOutputSearchResponse> {
+    return this.index.search<ResearchOutputSearchResponseEntity>(
+      query,
+      requestOptions,
```

Comment:
I can be wrong, but if we are doing this shouldn't we add the filter for the research-outputs entity here?
else nothing is stopping us from using this index to search for other entity types

Reply:
Yeah I don't think there anything that guarantees that we are searching for the right entity type

Reply:
we do